CRM/EventsExtras/Hook/Pre/Event.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+use CRM_EventsExtras_SettingsManager as SettingsManager;
+
+/**
+ * Class for Event Fee Pre Hook
+ */
+class CRM_EventsExtras_Hook_Pre_Event {
```

Comment:
I think we should name the class after what it does and not a generic name like `Event`. What about `EventDefaultSettings` or something better?

Reply:
I think we should have this call to handle all Event object?

Reply:
It's usually better to have the class describe what it is handling so it doesn't become a Jack of all class kindof. So it's better to name this after what it is doing. If we have another need, we can create a new class to handle that